Baseten provider

AlexKer opened the pull request:
This PR adds Baseten Model APIs with support of the DeepSeek, Llama and Kimi K2 models. We can now select Baseten as a provider in the Cline dropdown menu. This is similar to #4943.

This was tested locally with BASETEN_API_KEY and various unit tests (not included in PR).

-   [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
-   [x] ✨ New feature (non-breaking change which adds functionality)
-   [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] ♻️ Refactor Changes
-   [ ] 💅 Cosmetic Changes
-   [ ] 📚 Documentation update
-   [ ] 🏃 Workflow Changes

### Pre-flight Checklist

-   [x] Changes are limited to a single feature, bugfix or chore (split larger changes into separate PRs)
-   [x] Tests are passing (`npm test`) and code is formatted and linted (`npm run format && npm run lint`)
-   [x] I have created a changeset using `npm run changeset` (required for user-facing changes)
-   [x] I have reviewed [contributor guidelines](https://github.com/cline/cline/blob/main/CONTRIBUTING.md)

### Screenshots

Loom demo link: https://www.loom.com/share/acef33c66eca4a9494e947924fda79f0

### Additional Notes

pashpashpash commented:
Hey! Thanks for working on this @AlexKer! This is directionally correct, but looks like you got avalanched with merge conflicts. Do you mind resolving those and ping me for more thorough review + testing?

AlexKer commented:
@pashpashpash looks like a new provider `huaweiCloudMaas` was added since Monday, so ran into some conflicts. Should all be resolved now:)

arafatkatze commented on src/api/providers/baseten.ts:
```diff
@@ -0,0 +1,162 @@
+
+			// Handle usage information
+			if (chunk.usage) {
+				yield* this.yieldUsage(model.info, chunk.usage)
```
<img width="385" height="558" alt="image" src="https://github.com/user-attachments/assets/71d6260a-9ce9-4738-8637-05acaba775ce" />

The usage gets out on so many chunks my understanding is that it should come once at the end of the response.

For a simple hello message I see 797K tokens which is not possible. Fix the code here.

arafatkatze reviewed (changes requested):
Fix the token counting logic but emitting usage once

arafatkatze commented:
@AlexKer Your API is not persisted on the frontend. Please test the PR and fix the frontend and make sure you cna actually make a request that goes through

AlexKer commented:
@arafatkatze fixed, also added support for 2 new qwen3  models we added the past week. Here is a video showing it working: https://www.loom.com/share/ba73fcbe26e6477490852676bb096527?t=11&sid=2167adb0-880f-4b3e-beef-5ada27378e91

arafatkatze commented:
<img width="574" height="431" alt="image" src="https://github.com/user-attachments/assets/c86e3be7-6c24-4add-86f5-ecdc5672705a" />
These two models didn't actually support images

<img width="580" height="202" alt="image" src="https://github.com/user-attachments/assets/02980407-b717-407a-9baa-0a73b273de38" />

AlexKer commented:
@arafatkatze thanks for catching those! changed.

AlexKer commented:
I'm glad you enjoyed testing it -- much appreciated for your fast feedback. I can take responsibility in the accuracy based on our docs. We will look into a way of dynamically updating these values in the future.

arafatkatze commented:
Merged
